### PR TITLE
Feature fragments

### PR DESCRIPTION
**Pull request by zenangst**
This PR introduces fragments so that you can pass information between views that is not a part of the URNs.

### CODE REVIEW

**Comment by zenangst** [comment]
Need to fix the tests! 😁

File: CompassTests/Shared/TestCompass.swift [hunk 1]
```diff
@@ -24,7 +24,7 @@ class TestCompass: XCTestCase {
     let expectation = self.expectationWithDescription("Parse arguments")
     let url = NSURL(string: "compassTests://profile:testUser")!
 
-    Compass.parse(url) { route, arguments in
+    Compass.parse(url) { route, arguments, _ in
```
Comment (RamonGilabert) [inline CompassTests/Shared/TestCompass.swift:27]: Is there any way to get rid of this `_`? Maybe by setting it optional, or?

Comment (zenangst) [inline CompassTests/Shared/TestCompass.swift:27]: Not in a closure I'm afraid.

Comment (zenangst) [inline CompassTests/Shared/TestCompass.swift:27]: Well, you could make two methods for `parse`, like `Cartography` handles it but that feels a bit odd doesn't it?

Comment (RamonGilabert) [inline CompassTests/Shared/TestCompass.swift:27]: Well that depends on how often that third argument will be used. If it's never used I'd say copy the method and paste it. It feels a bit more weird to me to have it always as a `_`. But that's just a small thing, we can merge this one for now :)

Comment (RamonGilabert) [inline CompassTests/Shared/TestCompass.swift:27]: Waiting for Travis though.

Comment (zenangst) [inline CompassTests/Shared/TestCompass.swift:27]: I think you would always resolve it in your code but you don't need it in those tests, I added another test that actually does need the variable. The method that you call to navigate in your application can have it optional.

**Comment by RamonGilabert** [comment]
🍣

**Pull request merged**